monai/transforms/utility/dictionary.py

```diff
@@ -1419,6 +1423,36 @@ def __call__(self, data: Mapping[Hashable, NdarrayOrTensor]) -> dict[Hashable, N
         return d
 
 
+class TorchIOd(MapTransform, RandomizableTrait):
+    """
+    Dictionary-based wrapper of :py:class:`monai.transforms.TorchIO` for transforms.
+    All transforms in TorchIO can be applied randomly with probability p by specifying the `p=` argument.
+    """
+
+    backend = TorchIO.backend
+
+    def __init__(self, keys: KeysCollection, name: str, allow_missing_keys: bool = False, *args, **kwargs) -> None:
+        """
+        Args:
+            keys: keys of the corresponding items to be transformed.
+                See also: :py:class:`monai.transforms.compose.MapTransform`
+            name: The transform name in TorchIO package.
+            allow_missing_keys: don't raise exception if key is missing.
+            args: parameters for the TorchIO transform.
+            kwargs: parameters for the TorchIO transform.
+
+        """
+        super().__init__(keys, allow_missing_keys)
+        self.name = name
+        self.trans = TorchIO(name, *args, **kwargs)
+
+    def __call__(self, data: Mapping[Hashable, NdarrayOrTensor]) -> dict[Hashable, NdarrayOrTensor]:
+        d = dict(data)
+        for key in self.key_iterator(d):
+            d[key] = self.trans(d[key])
```

Comment:
So the TorchIO transform is instantiated once, i.e. the same object is used for all `d[key]`. 
As the randomness is handled per Transform, it is the same for all applications of the transform.

Reply:
But doesnt't the torchvision transform apply e.g. a different random flip each time it's called?

Reply:
So far I've only used it with subjects, for which the above statement is true.
In our most common use-case, you're absolutely right, as we're passing `torch.tensor`s and not `tio.subject`s by default.

tio handles that via the Subjects entity and by using the RandomTransform just to generate params that are then passed to a non-random version (e.g. `RandomAffine` samples params and then consturcts `Affine` ). The latter is actually applied to all images of a subject to achieve consistency.

If multiple `torch.tensor`s are passed, you're right, it's a different one for each `torch.tensor`. TorchIO is only able to apply the same transform if a `tio.Subject` is passed.

TorchVision samples the parameters directly in the forward call, i.e. applies the transform with the sampled parameters and does not store them. 

So I think the behavior is the same for both torchvision and torchio, that when using random transforms, different transform params are sampled and thus a transform with different params is applied.

To achieve consistent transforms per dict, I could add a flag, which then wraps the images in the dict into a `tio.Subject`.

Is this reasonable?

Reply:
If torchio provides this system, it would be nice to be able to use the subject or any other option to achieve this for the dictionary.
That sounds like a great option! (The 4Ds are channel+3spatial dims i'm assuming?) I'd make this flag true by default, since this is the behavior that is common in Monai Transforms.

I don't think this PR needs to fix that for torchvision, but you could add something about it in the docs (and maybe a warning if you want), so it's easy to spot in the future.